lib/jellyfish/component/hls/request_handler.ex

```diff
@@ -162,13 +188,24 @@ defmodule Jellyfish.Component.HLS.RequestHandler do
     end
   end
 
-  defp update_status(status, last_partial) do
+  defp wait_for_partial_ready(room_id, filename) do
+    GenServer.cast(registry_id(room_id), {:preload_hint, room_id, filename, self()})
+
+    receive do
+      :preload_hint_ready ->
+        EtsHelper.get_partial(room_id, filename)
+    end
+  end
+
+  defp update_status(status, last_partial, preload_hints) do
     {waiting_pids, status} =
       status
       |> Map.put(:last_partial, last_partial)
       |> pop_in([:waiting_pids, last_partial])
 
     send_partial_ready(waiting_pids)
+    send_preload_hint_ready(preload_hints)
+
```

Comment:
Either I am missing something, or the `:preload_hints` list in state is never emptied after sending the notifications...